src/prefect/transactions.py

```diff
@@ -58,6 +61,7 @@ class Transaction(ContextModel):
         default_factory=list
     )
     overwrite: bool = False
+    logger: Optional[PrefectLogAdapter] = None
```

Comment:
I assume we could handle any type of logger here
```suggestion
    logger: Optional[Logger] = None
```

Reply:
I tried that type hint, but it failed! Within the engine we use `task_run_logger` which apparently outputs a ... different type of object?

Reply:
That's whack. I guess this type hint would be the most technically correct then:
```suggestion
    logger: Union[logging.Logger, logging.LoggerAdapter, None] = None
```

Reply:
Begrudgingly updated lol